libs/community/langchain_community/chat_models/llama_edge.py

```diff
@@ -45,10 +53,26 @@ def _convert_message_to_dict(message: BaseMessage) -> dict:
     return message_dict
 
 
-class WasmChatService(BaseChatModel):
+def _convert_delta_to_message_chunk(
+    _dict: Mapping[str, Any], default_class: Type[BaseMessageChunk]
+) -> BaseMessageChunk:
+    role = _dict.get("role")
+    content = _dict.get("content") or ""
+
+    if role == "user" or default_class == HumanMessageChunk:
+        return HumanMessageChunk(content=content)
+    elif role == "assistant" or default_class == AIMessageChunk:
+        return AIMessageChunk(content=content)
+    elif role or default_class == ChatMessageChunk:
+        return ChatMessageChunk(content=content, role=role)
+    else:
+        return default_class(content=content)
+
+
+class ChatService(BaseChatModel):
```

Comment:
should this be named LlamaEdgeChatService to make it more specific? ChatService is kinda generic

Reply:
Sure! `LlamaEdgeChatService` sounds better. Will update the name. Thanks a lot!